src/config_file.c

```diff
@@ -851,34 +872,27 @@ static int read_on_variable(
 	return result;
 }
 
-static int config_read(
+static int config_read_buffer(
 	git_config_entries *entries,
 	const git_repository *repo,
 	git_config_file *file,
 	git_config_level_t level,
-	int depth)
+	int depth,
+	const char *buf,
+	size_t buflen)
 {
 	diskfile_parse_state parse_data;
 	git_config_parser reader;
-	git_buf contents = GIT_BUF_INIT;
 	int error;
```

Comment:
This needs to be initialized, as there's an error path uncovering it.
```
/Users/vsts/agent/2.153.2/work/1/s/src/config_file.c:898:6: error: variable 'error' is used uninitialized whenever 'if' condition is true [-Werror,-Wsometimes-uninitialized]
        if (!reader.ctx.content || *reader.ctx.content == '\0')
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/vsts/agent/2.153.2/work/1/s/src/config_file.c:910:9: note: uninitialized use occurs here
        return error;
               ^~~~~
```

Reply:
Yup, fixed!